server/internal/infprocessor/infprocessor.go

```diff
@@ -0,0 +1,136 @@
+package infprocessor
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+
+	v1 "github.com/llm-operator/inference-manager/api/v1"
+	"github.com/llm-operator/inference-manager/common/pkg/models"
+)
+
+const (
+	taskQueueSize  = 100
+	completionPath = "/v1/chat/completions"
+)
+
+// Task is an infeference task.
+type Task struct {
```

Comment:
not sure, wondering if we need to preserve  `context` also?

Reply:
good point! not sure. let me add a TODO.
```suggestion
// TODO(kenji): Consider preserving the request context as well.
type Task struct {
```